src/Cli/Microsoft.DotNet.Cli.Utils/PathUtility.cs

```diff
@@ -2,420 +2,418 @@
-                }
+                // on different volumes, "relative" path is just path2
+                return path2;
             }
-            else
-            {
-                compare = StringComparison.Ordinal;
-            }
-
-            var index = 0;
-            var path1Segments = path1.Split(Path.DirectorySeparatorChar, Path.AltDirectorySeparatorChar);
-            var path2Segments = path2.Split(Path.DirectorySeparatorChar, Path.AltDirectorySeparatorChar);
-            // if path1 does not end with / it is assumed the end is not a directory
-            // we will assume that is isn't a directory by ignoring the last split
-            var len1 = path1Segments.Length - 1;
-            var len2 = path2Segments.Length;
-
-            // find largest common absolute path between both paths
-            var min = Math.Min(len1, len2);
-            while (min > index)
-            {
-                if (!string.Equals(path1Segments[index], path2Segments[index], compare))
-                {
-                    break;
-                }
-                // Handle scenarios where folder and file have same name (only if os supports same name for file and directory)
-                // e.g. /file/name /file/name/app
-                else if ((len1 == index && len2 > index + 1) || (len1 > index && len2 == index + 1))
-                {
-                    break;
-                }
-                ++index;
-            }
-
-            var path = "";
+        }
+        else
+        {
+            compare = StringComparison.Ordinal;
+        }
 
-            // check if path2 ends with a non-directory separator and if path1 has the same non-directory at the end
-            if (len1 + 1 == len2 && !string.IsNullOrEmpty(path1Segments[index]) &&
-                string.Equals(path1Segments[index], path2Segments[index], compare))
+        var index = 0;
+        var path1Segments = path1.Split(Path.DirectorySeparatorChar, Path.AltDirectorySeparatorChar);
+        var path2Segments = path2.Split(Path.DirectorySeparatorChar, Path.AltDirectorySeparatorChar);
+        // if path1 does not end with / it is assumed the end is not a directory
+        // we will assume that is isn't a directory by ignoring the last split
+        var len1 = path1Segments.Length - 1;
+        var len2 = path2Segments.Length;
+
+        // find largest common absolute path between both paths
+        var min = Math.Min(len1, len2);
+        while (min > index)
+        {
+            if (!string.Equals(path1Segments[index], path2Segments[index], compare))
             {
-                return path;
+                break;
             }
-
-            if (includeDirectoryTraversals)
+            // Handle scenarios where folder and file have same name (only if os supports same name for file and directory)
+            // e.g. /file/name /file/name/app
+            else if (len1 == index && len2 > index + 1 || len1 > index && len2 == index + 1)
```

Comment:
I prefer the extraneous parentheses, as they make it easier to remember order of operations

Reply:
Hmm. I didn't change this.

Reply:
Again, if you don't want the IDE to do this, we need to determine:
- What analyzer is causing this
- What setting to change to disable it (if desired)

I can't keep making minor commits to every cleanup PR because the IDE automatically did some cleaning. 😵‍💫

Reply:
I've re-added the parentheses.

Reply:
I'm guessing [this one](https://github.com/dotnet/sdk/blob/8166b99ca3085adab75ed8e1117695b8fe66e076/.editorconfig#L354)